src/command-line/utils.js

```diff
@@ -51,6 +52,54 @@ class Utils {
 
 		return home;
 	}
+
+	// Parses CLI options such as `-c public=true`, `-c debug.raw=true`, etc.
+	static parseConfigOptions(val, memo) {
+		// Invalid option that is not of format `key=value`, do nothing
+		if (!val.includes("=")) {
+			return memo;
+		}
+
+		const parseValue = (value) => {
+			if (value === "true") {
+				return true;
+			} else if (value === "false") {
+				return false;
+			} else if (value === "undefined") {
+				return undefined;
+			} else if (value === "null") {
+				return null;
+			} else if (/^\[.*\]$/.test(value)) { // Arrays
```

Comment:
Could we just `JSON.parse` arrays and objects?

Reply:
I guess. The `if` would be the same though (otherwise I have to `try`/`catch` for strings, no thanks). I'll try with that, not sure if I like that yet. It's silly to re-write my own parser when `JSON.parse` exists, but I also don't want to give too much power to `-c`.

Also, I don't really want to support objects. Arrays are useful (for example for `transports`), but if you want to pass an object, use the dot-notation keys. I don't want to support everything and spend my life dealing with edge cases lol.

Reply:
I was thinking JSON.parse would be a single line/call compared to what you are doing now.

Reply:
You can't, because `JSON.parse` explodes on strings, except when they are `true`, `false`, or `null`. Ah, JavaScript...

Reply:
Alright, so I've played with this, and unfortunately I can't use `JSON.parse` for that for a few reasons:

- When passing `-c foo=true`, `JSON.parse(value)` correctly returns `true`. But if I pass `-c foo=bar`, it explodes. If I go the safe route and add quotes to all values, then ``JSON.parse(`"${value}"`)`` works for the latter, but the former is treated as `"true"` so I end up... having to re-check all special cases like `true`/`false`/etc. and even arrays lol (only I don't have to check for `null`, yay).
- I can't expect users to pass valid, quoted JSON for strings such as `-c foo="bar"` because, well first off it's lame and too easy to forget (programs never require that, but also  when I get the config in the code, quotes have been removed at shell level, so I get `foo=bar`. Back to square one. We might have to require users to do `-c foo=\"bar\"` which is just horrible.
- I was hoping to be able to do that for arrays at the very least but... `-c foo=[bar,baz]` has the exact same issue, `[bar,baz]` is not valid JSON.

So in the end, the only thing I can do is make sure all cases are correctly tested, which I think is what I'm doing :man_shrugging: Let me know if you think I forgot cases.